### PR TITLE
[App] Enable state broadcast with MultiNode

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

This PR enables broadcasting the state for local rank 0 within the multi-node components.

This is done by attaching the SetattributeProxy within the spawn process only on rank 0.

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

File: src/lightning_app/components/multi_node/pytorch_spawn.py
Comment (lantiga) [inline src/lightning_app/components/multi_node/pytorch_spawn.py:37]: let's add a method that removes the wrapper (and another that restores it), in case we need to do it in other places + the operations you need to do to remove the wrapper will change

It could actually be a context
```
with self.no_wrapper():
    ...
```
your call

**Review by lantiga — APPROVED** [review]
Looks great, just a suggestion. Adding a test is also recommended.

File: examples/app_multi_node/train_lite.py
Comment (tchaton) [reply examples/app_multi_node/train_lite.py:44]: removed.

File: src/lightning_app/components/multi_node/pytorch_spawn.py
Comment (tchaton) [reply src/lightning_app/components/multi_node/pytorch_spawn.py:37]: Done !

**Review by ethanwharris — APPROVED** [review]
LGTM, some manifest weirdness

File: MANIFEST.in
Comment (ethanwharris) [reply MANIFEST.in]: Oh, fixed already 😆

File: src/lightning_app/components/multi_node/pytorch_spawn.py
Comment (lantiga) [inline src/lightning_app/components/multi_node/pytorch_spawn.py:54]: Just an observation: here we are destroying WorkStateObserver when run returns, and eventually instantiate a new one on a subsequent call to run.
However, that run method might be starting up a thread that modifies the state and return immediately, and we would lose the ability to detect state changes resulting from that.
Wouldn’t it be better to initialize the observer during the first call to run, and then keep it running until the work is stopped?

File: src/lightning_app/utilities/proxies.py
Comment (lantiga) [inline src/lightning_app/utilities/proxies.py:558]: Same comment as above

**Review by lantiga — APPROVED** [review]
Approved to unblock for now but I have a concern regarding the lifetime of the WorkStateObserver (not for the training case but other cases)

File: src/lightning_app/components/multi_node/pytorch_spawn.py
Comment (tchaton) [reply src/lightning_app/components/multi_node/pytorch_spawn.py:54]: Yes, 100 %. I faced some concurrency issues in the past, but we can definitely re-explore it. This would definitely be the right way.

**Pull request merged**
